templates/repo/issue/view_content/add_reaction.tmpl

```diff
@@ -1,13 +1,13 @@
 {{if .ctxData.IsSigned}}
 <div class="item action ui dropdown jump pointing top right select-reaction" data-action-url="{{.ActionURL}}">
-	<a class="add-reaction">
+	<a role="button" class="add-reaction">
```

Comment:
It should be `<span>` here, no role.

This dropdown works like this: focus is on `<div "ui dropdown">` , see the aria.md

Reply:
`<span>`? Surely we don't do buttons as `<span>`, right?

Reply:
It's not a button. I guess I have mentioned this: "It should be `<span>` here, no role."

And, the `aria.md` has explained the problem, if it's not clear, I will improve.

Reply:
So we make it a `<div>` or `<span>` just to satisfy fomantic? To me, something that can be clicked should ideally be `<button>` or at least have `role=button`, do you think it would interfere if we make it that?

Reply:
> So we make it a <div> or <span> just to satisfy fomantic?

Not "satisfy", but each line of code should have its purpose.

>  something that can be clicked should ideally be <button> or at least have role=button, do you think it would interfere if we make it that?

1. It is not clickable, and it is not even focus-able, it has been described by `aria.md`.
2. The dropdown role has been provide by `.ui.dropdown`, by the aria patch framework.

So, what's the real purpose of making it `<a>` or `<button>` here?

Reply:
The click occurs on the `.ui.dropdown` itself, but not the inner element. By design, Fomantic UI only use this inner element for "display" purpose, it never wants to make it clickable.

Reply:
> The click occurs on the .ui.dropdown itself, but not the inner element

Ah, I didn't know that. In that case it makes sense to not have a inner button, but just a `<div>` or `<span>`. I assume Fomantic or our a11y script takes care so that the outer element is focusable etc?

Reply:
IIRC Fomantic UI itself makes the outer element focusable, our aria patch only helps about aria-xxx and role (and add a lot of `tabindex=-1` for menu items to prevent them from getting focus)

Reply:
We can't easily make this `<a>` something else because like 10 CSS selectors target the `a` tag name, and the "..." (context-menu) button in the comment header is also `<a>`. I did a few related cleanups by using `muted` style in place of some CSS rules, but I guess this cleanup is for another PR.